react/w-books/src/app/screens/SignUp/index.tsx

```diff
@@ -0,0 +1,114 @@
+  const [, setState] = useState({
+    email: '',
+    password: '',
+    passwordConfirmation: '',
+    firstName: '',
+    lastName: '',
+    locale: 'en'
+  });
+
+  const { register, errors, handleSubmit, watch } = useForm<User>();
+  const password = useRef({});
+  password.current = watch('password', '');
+
+  const onSubmit = (user: User): void => {
+    setState(user);
+  };
+
+  const handleLogin = (e: React.MouseEvent<HTMLElement>): void => {
+    e.preventDefault();
+    // Login code here
+  };
+
+  return (
+    <div className={styles.signupContainer}>
+      <img src={logo} alt="Logo" className={styles.signupLogo} />
+      <form onSubmit={handleSubmit(onSubmit)} className={styles.signupForm}>
+        <InputField
+          type="text"
+          label={i18next.t('SignUp:firstName')}
+          name={SIGNUP_FIELDS.firstName}
+          inputRef={register({
+            required: { value: true, message: i18next.t('SignUp:nameRequired') }
+          })}
+          error={errors.firstName}
+        />
+        <InputField
+          type="text"
+          label={i18next.t('SignUp:lastName')}
+          name={SIGNUP_FIELDS.lastName}
+          inputRef={register({
+            required: { value: true, message: i18next.t('SignUp:lastNameRequired') }
+          })}
+          error={errors.lastName}
+        />
+        <InputField
+          type="text"
+          label={i18next.t('SignUp:email')}
+          name={SIGNUP_FIELDS.email}
+          inputRef={register({
+            required: { value: true, message: i18next.t('SignUp:emailRequired') },
+            pattern: {
+              value: validations.emailPattern,
+              message: i18next.t('SignUp:invalidEmail')
+            }
+          })}
+          error={errors.email}
+        />
+        <InputField
+          type="password"
+          label={i18next.t('SignUp:password')}
+          name={SIGNUP_FIELDS.password}
+          inputRef={register({
+            required: { value: true, message: i18next.t('SignUp:passwordRequired') }
+          })}
+          error={errors.password}
+        />
+        <InputField
+          type="password"
+          label={i18next.t('SignUp:passwordConfirmation')}
+          name={SIGNUP_FIELDS.passwordConfirmation}
+          inputRef={register({
+            required: {
+              value: true,
+              message: i18next.t('SignUp:passwordConfirmationRequired')
+            },
+            validate: value => value === password.current || `${i18next.t('SignUp:passwordDoesNotMatch')}`
+          })}
+          error={errors.passwordConfirmation}
+        />
+        <button type="submit" className={styles.signupGreenButton}>
+          {i18next.t('SignUp:signUp')}
+        </button>
+      </form>
+
+      <button type="button" onClick={handleLogin} className={styles.signupGreyButton}>
+        {i18next.t('SignUp:login')}
+      </button>
+    </div>
+  );
+}
+
+export default SignUp;
```

Comment:
Check that the layout is not the same as the design. You have the labels and the input very near to each other. I think you should add some margin. Check that please